### PR TITLE
EPMRPP-34401 React. Launches. Modal.  Move to Debug

### PR DESCRIPTION
**Pull request by AmsterGet**

### CODE REVIEW

File: app/src/pages/inside/launchesPage/launchesPage.jsx [hunk 1]
```diff
@@ -170,6 +171,10 @@ export class LaunchesPage extends Component {
 
   mergeLaunches = () => this.props.mergeLaunchesAction(this.props.fetchData);
 
+  moveToDebugMultiple = () => {
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/launchesPage.jsx]: Done.

File: app/src/pages/inside/launchesPage/launchesPage.jsx [hunk 2]
```diff
@@ -189,6 +174,14 @@ export class LaunchesPage extends Component {
       fetchFunc: this.props.fetchData,
     });
 
+  moveToDebugLaunches = (eventData) => {
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/launchesPage.jsx]:
```
moveLaunchesToDebug = (eventData) => {
  const launches = eventData && eventData.id ? [eventData] : this.props.selectedLaunches;
  // ...
}
```

File: app/src/controllers/launch/actionCreators.js [hunk 3]
```diff
@@ -48,3 +48,13 @@ export const compareLaunchesAction = defineGroupOperation(
     }),
   () => null,
 );
+export const moveToDebugLaunchesAction = defineGroupOperation(
```
Comment (DiscoElevator) [inline app/src/controllers/launch/actionCreators.js]: moveLaunchesToDebugAction

File: app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx [hunk 4]
```diff
@@ -18,33 +24,93 @@ const messages = defineMessages({
...
     },
+    showNotification: () => {},
   };
-  confirmAndClose = (closeModal) => {
-    this.props.data.onConfirm();
+
+  moveAndClose = (closeModal) => {
+    const { ids, fetchFunc } = this.props.data;
+    const entities = {};
+    ids.forEach((id) => {
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx]: `const entities = ids.reduce((entities, id) => ({ ...entities, [id]: { mode: DEBUG.toUpperCase() }}), {} );`

File: app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx [hunk 5]
```diff
@@ -18,33 +24,93 @@ const messages = defineMessages({
...
+    });
+    fetch(this.props.url, {
+      method: 'put',
+      data: {
+        entities,
+      },
+    })
+      .then(() => {
+        const successMessage =
+          (ids.length > 1 &&
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx]: I think ternary operator will be more readable here

File: app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx [hunk 6]
```diff
@@ -55,7 +121,10 @@ export class MoveToDebugModal extends Component {
         okButton={okButton}
         cancelButton={cancelButton}
       >
-        <p className={cx('message')}>{Parser(intl.formatMessage(messages.moveToDebugText))}</p>
+        <p className={cx('message')}>
+          {(data.ids.length > 1 && Parser(intl.formatMessage(messages.moveToDebugMultipleText))) ||
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx]: I think ternary operator will be more readable here

File: app/src/pages/inside/launchesPage/launchesPage.jsx (hunk 2 above)
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/launchesPage.jsx]: Thank you for the suggestion. I will use a ternary operator more often
Fixed

File: app/src/controllers/launch/actionCreators.js (hunk 3 above)
Comment (AmsterGet) [reply app/src/controllers/launch/actionCreators.js]: Fixed

File: app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx (hunk 4 above)
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx]: Thank you for the suggestion. I didn't think about the reduce here.
Fixed

File: app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx (hunk 5 above)
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx]: Fixed

File: app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx (hunk 6 above)
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx]: Fixed

File: app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx [hunk 7]
```diff
@@ -18,33 +24,90 @@ const messages = defineMessages({
...
+      },
+    })
+      .then(() => {
+        const successMessage =
+          ids.length > 1
+            ? this.props.intl.formatMessage(messages.moveToDebugMultipleSuccessMessage)
+            : this.props.intl.formatMessage(messages.moveToDebugSuccessMessage);
+        this.props.showNotification({
+          message: successMessage,
+          type: 'success',
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx]: Please use notification types from 'controllers/notification'

File: app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx [hunk 8]
```diff
@@ -18,33 +24,90 @@ const messages = defineMessages({
...
+        });
+        fetchFunc();
+      })
+      .catch((response) => {
+        this.props.showNotification({
+          message: this.props.intl.formatMessage(messages.moveToDebugFailedMessage, {
+            message: response.message,
+          }),
+          type: 'error',
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx]: Please use notification types from 'controllers/notification'

File: app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx (hunk 7 above)
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx]: Fixed

File: app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx (hunk 8 above)
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx]: Fixed

**Pull request merged**